app/controllers/wpcc/your_contributions_controller.rb

```diff
@@ -9,16 +9,47 @@ def new
         session[:salary].to_i, session[:contribution_preference]
       )
 
+      store_eligible_salary
+
       @your_contributions_form = Wpcc::YourContributionsForm.new(
         employee_percent: @contribution.employee_percent,
         employer_percent: @contribution.employer_percent
       )
     end
 
+    def create
+      @your_contributions_form = Wpcc::YourContributionsForm.new(
+        convert_params
+      )
+
+      if @your_contributions_form.valid?
+        amend_session
+        redirect_to your_results_path
+      else
+        redirect_to new_your_contribution_path
+      end
+    end
+
     private
 
     def your_contributions_params
-      params.permit(:employee_percent, :employer_percent)
+      params.require(:your_contributions_form)
+            .permit(:employee_percent, :employer_percent)
+    end
+
+    def convert_params
```

Comment:
convert params to what? Should the aim of the conversion be expressed in the method name, perhaps?